experimental/dwarf.go

```diff
@@ -0,0 +1,22 @@
+package experimental
+
+import (
+	"context"
+)
+
+type enableDWARFBasedStackTraceKey struct{}
+
+// WithDWARFBasedStackTrace enables the DWARF based stack traces in the face of runtime errors.
+// This only takes into effect when the original Wasm binary has the DWARF "custom sections"
+// that are often stripped depending on the optimization options of the compilers.
+//
```

Comment:
Maybe add a small before/after (like 3 lines of a contrived stack trace), so people know what looks like before this.